### ISSUE TITLE
angular2-polyfills.js reruns 404 not found on Windows

### ISSUE THREAD

**Issue by kuncevich** [preboot/angular2-webpack#65]
I just updated my current project with this commit https://github.com/preboot/angular2-webpack/commit/4e797e4a058eba4de7f45fba80f86ccdc0263a4d and it is causing an issue `http://localhost:8080/polyfills/angular2-polyfills.js reruns 404 not found` which causing then this wired error chain:
```
Failed to load resource: the server responded with a status of 404 (Not Found) http://localhost:8080/polyfills/angular2-polyfills.js
localhost/:164 Uncaught reflect-metadata shim is required when using class decorators
localhost/:75 Uncaught TypeError: di_1.Injectable is not a function
```
Then just for test I `git clone` the skeleton to check if there is same thing when running clean and so when I run `npm start` got exactly same error.

**Comment by Foxandxss** [comment]
I cloned it, installed deps, run `npm start` and it works for me.

Are you sure you updated all bits? In concrete the task that copies the angular2-polyfills around.

**Comment by kuncevich** [comment]
Yes all updated. I just cloned the repo one more time and did clean `npm install` then `npm start` gave me the same error. Also tried in different browsers. BTW I am on Windows 10 may be it is making a difference and webpack is not working well in such case.

**Comment by kuncevich** [comment]
I can see an issue with absolute paths https://github.com/kevlened/copy-webpack-plugin/issues/26 reported. Could it be related?

**Comment by Foxandxss** [comment]
Uhmm, damn Windows.

Could you try to do a `npm run build` (which does the same thing as start but writes the files in the hard drive) and check if the file is created and more importantly **where**. Perhaps the path works different.

**Comment by Foxandxss** [comment]
Also, I hope I can remove this workaround as soon as the team makes the angular2-polyfill working good with an import.

**Comment by kuncevich** [comment]
After running `npm run build` got `dist` folder with all generated contend in it.
```
css\
img\
js\
polyfills\
index.html
service-worker.js
```
folder `polyfills` is actually contained `angular2-polyfills.js`

**Comment by Foxandxss** [comment]
That is not the output I wanted to see (it is the right one). So perhaps it is the dev server or something. Weeeird.

**Comment by kuncevich** [comment]
There is one more issue https://github.com/kevlened/copy-webpack-plugin/issues/29 that mentioned about files does not copy over when using webpack-dev-server

**Comment by Foxandxss** [comment]
So I assume that your "welcome page" was never having the big angular logo in it?

**Comment by kuncevich** [comment]
Actually I didn't have any issues with logo on home page

**Comment by Foxandxss** [comment]
More weird. Because it uses the same plugin to achieve the same thing.

I will set up node in my windows machine later and debug it myself.

**Comment by kuncevich** [comment]
OK thanks. Let me know if I can help or if you need me to provide any more details.

**Comment by Foxandxss** [comment]
After debugging it on windows, I think it is a bug on copy-webpack-plugin.

I will comment there and see what we can learn about that.

**Comment by yntelectual** [comment]
Hey, until the copy-webpack-plugin/webpack-dev-server behaviour is fixed, couldn't we bypass this issue by bundling the `angular2-polyfills.js` into a separate chunk that would be loaded first using `chunksSortMode` ? or what would happen if we require `angular2-polyfills.js` inside the `polyfills.js`?

**Comment by Foxandxss** [comment]
The upmost official word on this polyfills is that it needs to be loaded on a script tag, before anything else, including webpack.

Technically it doesn't support being required yet. This polyfills includes zone.js (which is angular's change detection system) and it needs to be loaded as soon as possible. If we create a new chunk with just the polyfills, we could potentially make some instances of our app to run outside zone.js and by that, no change detection.

That being said, it could potentially not happen but there is the risk.

I had to put a different solution in here which makes this risk a possibility.

If we manage to land the fix on the copy plugin I would be willing to reconsider it again.

That unless you have a better solution than copy plugin that would make the polyfills run as soon as possible.

Cheers.

**Comment by ocombe** [comment]
If we make a new entry, it will be added to the index file as a script tag

**Comment by Foxandxss** [comment]
But wouldn't that involve webpack first?

Like loading -> webpack -> polyfills -> rest

All things considered, the polyfills is already a webpack UMD bundle.

My biggest issue here is that I write the angular 2 docs and I need to document webpack for it and that is where the official word becomes *my word* so I don't want to say something wrong.

**Comment by Foxandxss** [comment]
Now that I am thinking. Webpack is not really involved in runtime, so having an entry point with the polyfills (that is a UMD bundle already) should make no real difference, shouldn't it? Grrr, too many questions.

**Comment by yntelectual** [comment]
This should be fixed un version `2.0.1`  of copy plugin.

**Comment by Foxandxss** [comment]
Yeah, thanks for the fix, but I don't need to use the plugin anymore for this. The actual solution works better.
